runtime/common/src/main/kotlin/kotlinx/serialization/context/SerialModules.kt

```diff
@@ -0,0 +1,168 @@
+     */
+    fun registerIn(context: MutableSerialContext)
+}
+
+/**
+ * A [SerialModule] which registers one class with one serializer for [ContextSerializer].
+ *
+ * @see MutableSerialContext.registerSerializer
+ */
+class SimpleModule<T: Any>(val kClass: KClass<T>, val kSerializer: KSerializer<T>): SerialModule {
+    override fun registerIn(context: MutableSerialContext) {
+        context.registerSerializer(kClass, kSerializer)
+    }
+}
+
+/**
+ * A [SerialModule] which registers multiple classes with its serializers for [ContextSerializer].
+ */
+@Suppress("UNCHECKED_CAST")
+class MapModule(val map: Map<KClass<*>, KSerializer<*>>): SerialModule {
+    override fun registerIn(context: MutableSerialContext) {
+        map.forEach { (k, s) -> context.registerSerializer(k as KClass<Any>, s as KSerializer<Any>) }
+    }
+}
+
+/**
+ * A [SerialModule] for composing other modules
+ *
+ * Has convenient operator [plusAssign].
+ *
+ * @see SerialModule.plus
+ */
+class CompositeModule(modules: List<SerialModule> = listOf()): SerialModule {
+    constructor(vararg modules: SerialModule) : this(modules.toList())
+
+    private val modules: MutableList<SerialModule> = modules.toMutableList()
+
+    override fun registerIn(context: MutableSerialContext) {
+        modules.forEach { it.registerIn(context) }
+    }
+
+    public operator fun plusAssign(module: SerialModule): Unit { modules += module }
+    public fun addModule(module: SerialModule) = plusAssign(module)
+}
+
+/**
+ * Composes [this] module with [other].
+ */
+operator fun SerialModule.plus(other: SerialModule): CompositeModule {
+    if (this is CompositeModule) {
+        this += other
+        return this
+    }
+    return CompositeModule(this, other)
+}
+
+/**
+ * A [SerialModule] which registers all its content for polymorphic serialization in the scope of [baseClass].
+ * If [baseSerializer] is present, registers it as a serializer for [baseClass] (which is useful if base class is serializable).
+ * Subclasses with its serializers can be added via [addSubclass] or [unaryPlus].
+ *
+ * @see PolymorphicSerializer
+ * @see MutableSerialContext.registerPolymorphicSerializer
+ */
+@Suppress("UNCHECKED_CAST")
+class PolymorphicModule<Base : Any>(val baseClass: KClass<Base>, val baseSerializer: KSerializer<Base>? = null) : SerialModule {
+    private val subclasses: MutableList<Pair<KClass<out Base>, KSerializer<out Base>>> = mutableListOf()
+    override fun registerIn(context: MutableSerialContext) {
+        if (baseSerializer != null) context.registerPolymorphicSerializer(baseClass, baseClass, baseSerializer)
+        subclasses.forEach { (k, s) ->
+            context.registerPolymorphicSerializer(
+                baseClass,
+                k as KClass<Base>,
+                s as KSerializer<Base>
+            )
+        }
+    }
+
+    /**
+     * @see PolymorphicModule.unaryPlus
+     */
+    public fun <T : Base> addSubclass(subclass: KClass<T>, serializer: KSerializer<T>) {
```

Comment:
add -> register to keep it consistent

Reply:
Also, it is worth to add reified version of `addSubclass` with `ImplicitReflectionSerializer`

Reply:
While we register serializers in _contexts_, this is a _module_, which is defined as a 'collection of serializers', and `add` is a natural verb for collection. It also emphasises that actual registering in some context would only be performed after module installation.